### PR TITLE
Fix term: leader only increases term on election_timeout

### PR DESCRIPTION
**Pull request by Parsifal22**
Changed the logic of the terms, and also changed the tests

### CODE REVIEW

**Review by maksimdrachov — CHANGES_REQUESTED** [review]

File: tests/raft_node.py [hunk 1]
```diff
@@ -386,59 +388,59 @@ async def _unittest_raft_node_append_entries_rpc() -> None:
...
-      | 9          |     Log term
+      | 2          |     Log term
       | top_4 <= 14|     Name <= value
       |____________|
 
      Step 7: Try to append valid log entry, however entry at prev_log_index term does not match
        ____________
       | 4          |     Log index
-      | 11         |     Log term
+      | 4          |     Log term
```
[the remark this answers is not in the document]
Comment (Parsifal22) [reply tests/raft_node.py:443]: Yeah, You are absolutely right

File: tests/raft_log_replication.py [hunk 2]
```diff
@@ -313,7 +313,7 @@ async def _unittest_raft_log_replication() -> None:
 
     for index, new_entry in enumerate(new_entries):
         request = sirius_cyber_corp.AppendEntries_1.Request(
-            term=raft_node_1._term,
+            term=1,
```
[the remark this answers is not in the document]
Comment (Parsifal22) [reply tests/raft_log_replication.py]: Fixed in commit [5cecfa8](https://github.com/OpenCyphal-Garage/cyraft/pull/18/commits/5cecfa80d1628e192be517d795f2645333c15472)

File: tests/raft_log_replication.py [hunk 3]
```diff
@@ -332,53 +332,53 @@ async def _unittest_raft_log_replication() -> None:
...
-        term=9,
+        term=1,
         entry=sirius_cyber_corp.Entry_1(
             name=uavcan.primitive.String_1(value="top_3"),
             value=12,
         ),
     )
     request = sirius_cyber_corp.AppendEntries_1.Request(
-        term=raft_node_1._term,
+        term=1,
```
[the remark this answers is not in the document]
Comment (Parsifal22) [reply tests/raft_log_replication.py]: fixed in commit [5cecfa8](https://github.com/OpenCyphal-Garage/cyraft/pull/18/commits/5cecfa80d1628e192be517d795f2645333c15472)

File: tests/raft_node.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Parsifal22) [reply tests/raft_node.py]: fixed in commit [5cecfa8](https://github.com/OpenCyphal-Garage/cyraft/pull/18/commits/5cecfa80d1628e192be517d795f2645333c15472)

**Review by maksimdrachov — APPROVED** [review]

**Pull request merged**